### PR TITLE
add filter and sort functionality

### PR DESCRIPTION
**Pull request by webdevelukas**
- Add possibility to filter matches by `gameId` and/or `username`
_Example: /matches?page=5&itemsPerPage=2&gameId=`gameId`&username=`username`_
- Matches are now sorted from the newest to the oldest

### CODE REVIEW

File: src/lib/router.ts [hunk 1]
```diff
@@ -85,12 +85,38 @@ type PaginatedMatches = {
   results: Match[];
 };
 
+type FilterProps = { gameId?: number; username?: string };
+
+function getFilter(filterOptions: FilterProps) {
```
Comment (webdevelukas) [inline src/lib/router.ts:90]: Maybe this function should be called `getMatchesFilter` and been placed in the `matches.ts` file.

File: src/lib/router.ts [hunk 2]
```diff
@@ -85,12 +85,38 @@ type PaginatedMatches = {
...
+  const { gameId, username } = filterOptions;
+  const filter: FilterProps = {};
+
+  if (typeof gameId !== 'undefined') {
+    filter.gameId = Number(gameId);
+  }
+  if (typeof username !== 'undefined') {
+    filter.username = String(username);
+  }
```
Comment (AlexisRoe) [inline src/lib/router.ts:99]: What happens if the values are arrays?
```suggestion
  if (typeof gameId !== 'undefined' && !Array.isArray(gameId)) {
    filter.gameId = Number(gameId);
  }
  if (typeof username !== 'undefined' && !Array.isArray(username)) {
    filter.username = String(username);
  }
```

**Review by AlexisRoe — CHANGES_REQUESTED** [review]
easy solution

**Comment by webdevelukas** [comment]
Closed in favor of changes in #11

**Pull request closed without merging**